### ISSUE TITLE
LLM2CLIP training

### ISSUE THREAD

**Issue by LKELN** [microsoft/LLM2CLIP#30]
When I was training the model, I found that the model's parameters were not being updated and the accuracy on the validation set remained unchanged. Even when I changed the learning rate, there was still no variation. Moreover, the model would encounter errors every time it reached the second epoch. I first used LLm2clip 8B to extract all text features for use during training. The visual model used is EVA02-CLIP-L-14-336. The training parameters are as follows.   OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=4 --master_port 16323 \
	--use_env training/main.py \
        --enable-deepspeed \
        --grad-checkpointing \
        --save-frequency 1  \
        --grad_accumulation_steps 4 \
        --report-to="tensorboard" \
        --wandb-project-name="LLM2CLIP" \
        --wandb-notes="EVA02-CLIP-L-14-336" \
        --pretrained=${PRETRAINED} \
        --dataset-resampled \
        --precision "fp16" \
        --warmup 0 \
        --batch-size=256 \
        --eval-batch-size=256 \
        --log-every-n-steps 50 \
        --epochs=5 \
        --lr=1e-5 \
        --visual-lr=1e-2 \
        --text-lr=1e-2 \
        --wd=0.05 \
        --visual-wd=0.05 \
        --text-wd=0.05 \
        --ld=1.0 \
        --text-ld=1.01 \
        --visual-ld=0.85 \
        --grad-clip-norm=5.0 \
        --smoothing=0. \
        --workers=8 \
        --model=${MODEL} \
        --seed 4096 \
        --gather-with-grad \
        --local-loss \
        --force-custom-clip \
        --optimizer="ap_adamw" \
        --zero-stage=1 \

**Comment by LKELN** [comment]
Here is the erro: Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data.
[E ProcessGroupNCCL.cpp:495] To avoid data inconsistency, we are taking the entire process down.
[E ProcessGroupNCCL.cpp:916] [Rank 2] NCCL watchdog thread terminated with exception: [Rank 2] Watchdog caught collective operation timeout: WorkNCCL(SeqNum=175, OpType=ALLREDUCE, NumelIn=1, NumelOut=1, Timeout(ms)=1800000) ran for 1800251 milliseconds before timing out.
terminate called after throwing an instance of 'std::runtime_error'

**Comment by LKELN** [comment]
For the first problem, it can be solved by changing the value of --grad-accumulation-steps to 1.

**Comment by wswaq** [comment]
For the first problem, since grad_accumulation_steps does not increase the number of negative samples in contrastive learning, we did not use it. There might be some bugs related to this.

For the second problem, did the timeout occur during the model evaluation? We have encountered timeouts during evaluation before. If possible, please provide more details.

**Comment by LKELN** [comment]
I've already solved it. It seems to be the problem brought by grad_accumulation_steps.
